src/controllers/auth.ts

```diff
@@ -0,0 +1,235 @@
+import { NextFunction, Request, Response } from "express";
+import userModel, { IUser } from "../models/user";
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
+import { Document } from "mongoose";
+import { refreshTokenExpires, tokenExpires, tokenSecret } from "../config";
+
+// eslint-disable-next-line max-statements
+const register = async (req: Request, res: Response) => {
```

Comment:
Try to make these controller functions be an `AuthController` that inherits from `BaseController<User>` to reuse code and keep the code in the same style.
You can override methods to customize their behavior.
You can add methods that are not CRUD such as login and logout.